### PR TITLE
Migrate Azure Storage Queues Binding to Track2 SDK

### PR DESCRIPTION
**Pull request by berndverst**
# Description
Migrate Azure Storage Queues Binding to Track2 SDK

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: bindings/azure/storagequeues/storagequeues.go [hunk 1]
```diff
@@ -51,50 +51,64 @@ type QueueHelper interface {
...
-		URL, _ := url.Parse(fmt.Sprintf("https://%s.queue.%s/%s", m.AccountName, env.StorageEndpointSuffix, m.QueueName))
-		d.queueURL = azqueue.NewQueueURL(*URL, p)
+		credential, tokenErr := azEnvSettings.GetTokenCredential()
+		if tokenErr != nil {
+			return nil, fmt.Errorf("invalid token credentials with error: %w", tokenErr)
+		}
+		var clientErr error
+		queueServiceClient, clientErr = azqueue.NewServiceClient(m.GetQueueURL(azEnvSettings), credential, &options)
+		if clientErr != nil {
+			return nil, fmt.Errorf("cannot init storage queue client with Azure AD token: %w", err)
```
Comment (ItalyPaleAle) [inline bindings/azure/storagequeues/storagequeues.go:101]:
```suggestion
			return nil, fmt.Errorf("cannot init storage queue client with Azure AD token: %w", clientErr)
```

File: bindings/azure/storagequeues/storagequeues.go [hunk 2]
```diff
@@ -51,50 +51,64 @@ type QueueHelper interface {
...
 	}
 
+	d.decodeBase64 = m.DecodeBase64
+	d.encodeBase64 = m.EncodeBase64
+	d.visibilityTimeout = *m.VisibilityTimeout
+	d.queueClient = queueServiceClient.NewQueueClient(m.QueueName)
+
 	createCtx, createCancel := context.WithTimeout(ctx, 2*time.Minute)
-	_, err = d.queueURL.Create(createCtx, azqueue.Metadata{})
+	_, err = d.queueClient.Create(createCtx, &azqueue.CreateOptions{})
```
Comment (ItalyPaleAle) [inline bindings/azure/storagequeues/storagequeues.go:111]:
```suggestion
	_, err = d.queueClient.Create(createCtx, nil)
```

File: bindings/azure/storagequeues/storagequeues.go [hunk 3]
```diff
@@ -119,37 +136,41 @@ func (d *AzureQueueHelper) Write(ctx context.Context, data []byte, ttl *time.Dur
...
+		VisibilityTimeout: ptr.Of(int32(d.visibilityTimeout.Seconds())),
+	})
 	if err != nil {
 		return err
 	}
-	if res.NumMessages() == 0 {
+	if len(res.Messages) == 0 {
 		// Queue was empty so back off by 10 seconds before trying again
 		select {
 		case <-time.After(10 * time.Second):
```
Comment (ItalyPaleAle) [inline bindings/azure/storagequeues/storagequeues.go:157]: Should this be configurable?

File: bindings/azure/storagequeues/storagequeues.go [hunk 4]
```diff
@@ -119,37 +136,41 @@ func (d *AzureQueueHelper) Write(ctx context.Context, data []byte, ttl *time.Dur
...
 		return nil
 	}
-	mt := res.Message(0).Text
+	mt := res.Messages[0].MessageText
 
 	var data []byte
 
 	if d.decodeBase64 {
-		decoded, decodeError := base64.StdEncoding.DecodeString(mt)
+		decoded, decodeError := base64.StdEncoding.DecodeString(*mt)
```
Comment (ItalyPaleAle) [inline bindings/azure/storagequeues/storagequeues.go:167]: Does this need a nil-check? Also below on `data = []byte(*mt)`

File: bindings/azure/storagequeues/storagequeues.go [hunk 5]
```diff
@@ -159,9 +180,8 @@ func (d *AzureQueueHelper) Read(ctx context.Context, consumer *consumer) error {
 	if err != nil {
 		return err
 	}
-	messageIDURL := messagesURL.NewMessageIDURL(res.Message(0).ID)
-	pr := res.Message(0).PopReceipt
-	_, err = messageIDURL.Delete(ctx, pr)
+
+	_, err = d.queueClient.DeleteMessage(ctx, *res.Messages[0].MessageID, *res.Messages[0].PopReceipt, &azqueue.DeleteMessageOptions{})
```
Comment (ItalyPaleAle) [inline bindings/azure/storagequeues/storagequeues.go:184]:
```suggestion
	_, err = d.queueClient.DeleteMessage(ctx, *res.Messages[0].MessageID, *res.Messages[0].PopReceipt, nil)
```

File: bindings/azure/storagequeues/storagequeues.go [hunk 6]
```diff
@@ -159,9 +180,8 @@ func (d *AzureQueueHelper) Read(ctx context.Context, consumer *consumer) error {
 	if err != nil {
 		return err
 	}
-	messageIDURL := messagesURL.NewMessageIDURL(res.Message(0).ID)
-	pr := res.Message(0).PopReceipt
-	_, err = messageIDURL.Delete(ctx, pr)
+
```
Comment (ItalyPaleAle) [inline bindings/azure/storagequeues/storagequeues.go:183]: I think .MessageID etc need a nil-check

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Some small tweaks but overall LGTM

**Comment by ItalyPaleAle** [comment]
In a separate PR, can you also remove from the internal auth package all code related to track 1? I don't think it's needed anymore after this! 🎉

File: bindings/azure/storagequeues/storagequeues.go (hunk 3 above)
Comment (berndverst) [reply bindings/azure/storagequeues/storagequeues.go:157]: Maybe - but this PR is just the SDK upgrade

Comment (ItalyPaleAle) [reply bindings/azure/storagequeues/storagequeues.go:157]: A separate issue/PR is fine!

**Comment by berndverst** [comment]
> In a separate PR, can you also remove from the internal auth package all code related to track 1? I don't think it's needed anymore after this! 🎉

One of us can do it - doesn't have to be necessarily, but I'll do it if I remember -- next week!

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**